README.md

```diff
@@ -1,38 +1,146 @@
+
 <!-- omit in toc -->
-# Clash Ethernet Core
-TODO: Describe features
+# Clash Ethernet 🙏✨
+A fully configurable Ethernet core written in [Clash](https://clash-lang.org/).
 
 <!-- omit in toc -->
 # Table of Contents
-- [Working with this project](#working-with-this-project)
-- [REPL](#repl)
+- [Introduction](#introduction)
+- [Comparison with Other Ethernet Stacks](#comparison-with-other-ethernet-stacks)
+- [Resource Usage Comparison](#resource-usage-comparison)
+- [Documentation](#documentation)
+- [How to use as a developer](#how-to-use-as-a-developer)
+    - [Running Python tests](#running-python-tests)
+- [How to contact maintainers/developers](#how-to-contact-maintainersdevelopers)
+- [Acknowledgements](#acknowledgements)
+
+# Introduction
+Clash Ethernet is a fully configurable Ethernet core written in Clash.
+It aims to become part of the
+[clash-protocols](https://github.com/clash-lang/clash-protocols) library
+which exists to make it easy to develop and use on-chip communication
+protocols, with a focus on protocols in need of bidirectional
+communication. Clash Ethernet will provide the internet protocols.
+
+So far, Clash Ethernet implements the following internet protocols:
+- Ethernet
+- ICMP
+- IP
+- ARP
+
+# Comparison with Other Ethernet Stacks
+
+| Feature                  | Clash Ethernet                           | Verilog            | LiteEth                         |
+|--------------------------|:----------------------------------------:|:------------------:|:-------------------------------:|
+| Data width<br>(in bytes) | :white_check_mark:<br>Fully configurable | :x:<br>1 or 8      | :x:<br>1, 2, 4 or (partially) 8 |
+| ARP                      | :white_check_mark:                       | :white_check_mark: | :white_check_mark:              |
+| DHCP                     | :x:                                      | :x:                | :white_check_mark:              |
+| ICMP(echo)               | :white_check_mark:                       | :x:                | :white_check_mark:              |
+| IP                       | :white_check_mark:                       | :white_check_mark: | :white_check_mark:              |
+| UDP                      | :x:                                      | :white_check_mark: | :white_check_mark:              |
+
+Some of the benefits of using Clash Ethernet are:
+- Every protocol is fully configurable in data width. This is
+  something that other libraries, such as
+  [LiteEth](https://github.com/enjoy-digital/liteeth) and
+  [verilog-ethernet](https://github.com/alexforencich/verilog-ethernet)
+  do not support.
+
+- Clash makes it very easy to combine two or more components, see
+  [Clash Protocols](https://github.com/clash-lang/clash-protocols).
+  For example:
+
+  ```haskell
+  verifyChecksum |> depacketizerC const |> verifyLength
```

Comment:
maybe show some circuit notation with a nonlinear circuit? This is oficially a Protocols feature but it is definitely a merit of using our core